crates/goose-mcp/src/developer/mod.rs

```diff
@@ -719,3 +719,308 @@ impl Clone for DeveloperRouter {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use serde_json::json;
+    use tokio::sync::OnceCell;
+
+    static DEV_ROUTER: OnceCell<DeveloperRouter> = OnceCell::const_new();
+
+    async fn get_router() -> &'static DeveloperRouter {
+        DEV_ROUTER
+            .get_or_init(|| async { DeveloperRouter::new() })
+            .await
+    }
+
+    #[tokio::test]
+    async fn test_shell_missing_parameters() {
+        let router = get_router().await;
+        let result = router.call_tool("shell", json!({})).await;
+
+        assert!(result.is_err());
+        let err = result.err().unwrap();
+        assert!(matches!(err, ToolError::InvalidParameters(_)));
+    }
+
+    #[tokio::test]
+    async fn test_shell_change_directory() {
+        let router = get_router().await;
+        let result = router
+            .call_tool("shell", json!({ "working_dir": ".", "command": "pwd" }))
```

Comment:
we don't support working_dir anymore? (surprised this test passes?)

Reply:
ahhh let me clean that up, looks like it just ignores that argument and we happen to already be in `.` so it passes